app/crud/storage.py

```diff
@@ -0,0 +1,42 @@
+from typing import Generator, Iterator, BinaryIO
+
+from minio import S3Error
+from minio.datatypes import Object
+from minio.helpers import ObjectWriteResult
+
+from app.config import minio_settings
+from app.storage import storage
+
+
+def is_file_exists(path_to_object: str) -> bool:
+    try:
+        if storage.stat_object(minio_settings.BUCKET_NAME, path_to_object) != None:
+            return True
+        else:
+            return False
+    except S3Error as e:
+        return False
+
+
+def get_file_stream(full_path: str) -> Generator[bytes, any, None]:
+    file_response = storage.get_object(minio_settings.BUCKET_NAME, full_path)
+    async def __file_stream_generator():
+        try:
+            for chunk in file_response.stream():
+                yield chunk
+        finally:
+            file_response.close()
+            file_response.release_conn()
+    return __file_stream_generator()
```

Comment:
Этот код можно заменить на этот? Или нам явно нужно закрывать соединение?

```suggestion
def get_file_stream(full_path: str) -> Generator[bytes, any, None]:
    async def __file_stream_generator():
        with storage.get_object(minio_settings.BUCKET_NAME, full_path) as file_response:
            for chunk in file_response.stream():
                    yield chunk
    return __file_stream_generator()
```

Reply:
Да, нужно

Reply:
Тогда, если sdk не предоставляет контекстного менеджера я бы посоветовал написать его самим, чем писать каждый раз try/finally с закрытием соединения.